test/Conversion/LinalgToTpp/linalg-to-tpp-memref.mlir

```diff
@@ -679,3 +679,23 @@ func.func @linalg_fill_3d(%arg0: memref<2x8x32xf32>) -> memref<2x8x32xf32> {
   linalg.fill ins(%cst : f32) outs(%arg0 : memref<2x8x32xf32>)
   return %arg0 : memref<2x8x32xf32>
 }
+
+// -----
+
+#map = affine_map<(d0, d1) -> ()>
+#map1 = affine_map<(d0, d1) -> (d0, d1)>
+
+// CHECK-LABEL: scalar_input
+func.func @scalar_input(%arg0: memref<f32>, %arg1: memref<4x4xf32>) {
+  // CHECK-NOT: tpp.add
+  linalg.generic {
+    indexing_maps = [#map, #map1],
+    iterator_types = ["parallel", "parallel"]} 
+    ins(%arg0: memref<f32>)
```

Comment:
Here, your input is `memref<f32>` but in the comment you say `SCALAR means f32`.

You don't have enough tests to make sure all shapes are understood.

Reply:
SCALAR means f32, not `memref<f32>`. The latter is not handled.

Reply:
In your implementation, yes. My question is why are we doing this?

Why is it not handled?

Reply:
the support of memref<f32> never got implemented. Probably "we" was me at the beginning.

Reply:
This is not clear to me what you mean and what support we're talking about. It will depend on what input IR we have, so the question is: where is this coming from and how can we support the types we need without adding more ops to the IR.

Reply:
What the spec allows or not depends on whether they appear in the wild or not. If they do, it would be simpler to support them than to force canonicalization.

But without knowing where this problem is coming from, it's hard to tell if the spec needs changing or not.